apps/prairielearn/src/pages/studentGradebook/studentGradebook.html.tsx

```diff
@@ -56,42 +55,44 @@ export function StudentGradebook({
         <table class="table table-sm table-hover" aria-label="Gradebook">
           <thead>
             <tr>
-              <th style="width: 1%"><span class="visually-hidden">Label</span></th>
-              <th><span class="visually-hidden">Title</span></th>
+              <th style="width: 1%">
+                <span class="visually-hidden">Label</span>
+              </th>
+              <th>
+                <span class="visually-hidden">Title</span>
+              </th>
               <th class="text-center">Score</th>
             </tr>
           </thead>
           <tbody>
-            ${rows.map(
-              (row) => html`
-                ${row.start_new_set
-                  ? html`
-                      <tr>
-                        <th colspan="3">${row.assessment_set_heading}</th>
-                      </tr>
-                    `
-                  : ''}
+            {rows.map((row) => (
+              <Fragment key={row.assessment_set_heading}>
```

Comment:
I don't think this is an appropriate key since it's not unique per row. When considering an appropriate key, remember that some assessments may have multiple instances (e.g. "Auto-graded randomized practice exams" in the example course).

Something like `${assessment.id}-${assessment_instance.id}` might be good (where the instance ID is possibly `null`).